datasource/threedi_results.py

```diff
@@ -297,6 +326,20 @@ def aggregate_result_admin(self):
         file_like_object_nc = open(agg_path, 'rb')
         return GridH5AggregateResultAdmin(file_like_object_h5, file_like_object_nc)
 
+    @cached_property
+    def water_quality_result_admin(self):
+        try:
+            # Note: both of these might raise the FileNotFoundError
+            wq_path = find_water_quality_netcdf(self.file_path)
+            h5 = self.h5_path
+        except FileNotFoundError:
+            logger.exception("Water quality result not found")
+            return None
+        file_like_object_h5 = open(h5, 'rb')
+        file_like_object_h5.startswith = lambda x: False
+        file_like_object_nc = open(wq_path, 'rb')
```

Comment:
O, I still remember, this has to do with an issue in threedigrid - maybe we should add a comment pointing to the issue: https://github.com/nens/threedigrid/issues/183 ? I may might have forgotten this in the future :-(

Reply:
Added comments